internal/run/source.go

```diff
@@ -335,31 +337,19 @@ func (w *Workflow) snapshotSource(ctx context.Context, parentStep *workflowTrack
 			log.From(ctx).Warnf("error parsing registry location %s: %v", string(source.Registry.Location), err)
 		}
 
-		if orgSlug != auth.GetOrgSlugFromContext(ctx) {
-			message := fmt.Sprintf("current authenticated org %s does not match provided location %s", auth.GetOrgSlugFromContext(ctx), string(source.Registry.Location))
-			if !env.IsGithubAction() {
-				message += " run `speakeasy auth logout`"
-			}
-			if speakeasySelf == auth.GetOrgSlugFromContext(ctx) && !env.IsGithubAction() {
-				log.From(ctx).Warn(message)
-			} else {
-				return fmt.Errorf(message)
-			}
+		skip, key, err := getAndValidateAPIKey(ctx, orgSlug, workspaceSlug, string(source.Registry.Location))
+
+		if skip {
+			registryStep.Skip("you are authenticated with speakeasy-self")
+			return nil
 		}
 
-		if workspaceSlug != auth.GetWorkspaceSlugFromContext(ctx) {
```

Comment:
Shouldn't we still have an error message like this if you are not in speakeasy-self and you don't have the correct workspace authenticated